.github/workflows/android-build-and-test.yml

```diff
@@ -0,0 +1,98 @@
+#
+# Copyright 2024 Adobe. All rights reserved.
+# This file is licensed to you under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License. You may obtain a copy
+# of the License at http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+# OF ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+#
+
+name: Build and Test (Android)
+
+on:
+  workflow_call:
+    inputs:
+      android-api-levels:
+        description: |
+          The Android API levels to use for testing. Must be formatted as a valid JSON array string.
+          Example: '[29, 30]'
+            - Notice the use of single quotes to wrap the JSON array string, and number format for values.
+        type: string
+        default: '[29]'
+      # Flags for which tests to run
+      run-test-unit:
+        type: boolean
+        default: false
+      run-test-functional:
+        type: boolean
+        default: false
+      run-test-integration:
+        type: boolean
+        default: false
+      run-build-app:
+        type: boolean
+        default: false
+      enable-codecov:
+        description: 'Enable Codecov for test coverage.'
+        type: boolean
+        default: true
```

Comment:
nit: either we default everything to false or think about flipping some other defaults to true like unit and functional which we usually want to run for every build.

Reply:
Sounds good!